### ISSUE TITLE
Abide validateText (using custom pattern) never returns true

### ISSUE THREAD

**Issue by mheydman** [zurb/foundation-sites#8157]
In Foundation 6.1.1, I have added a custom pattern like this:
`Foundation.Abide.defaults.patterns['p_username'] = /^\w{4,16}$/;`

My form field was defined like so:
`<input type="text" id="nick" name="Nick" required pattern="p_username">`

The field is never considered valid. When I step through the Abide JS, this statement always returns false (line 336 of foundation.abide.js, I have removed comments from the code displayed here): 
```
return inputText.length ?
            this.options.patterns.hasOwnProperty(pattern) ? this.options.patterns[pattern].test(inputText) :
            pattern && pattern !== $el.attr('type') ? new RegExp(pattern).test(inputText) : true : true;
```

It seems that "new RegExp(pattern)" is evaluating to "new RegExp(p_username)" instead of "new RegExp(/^\w{4,16}$/)"

I believe that changing 
`new RegExp(pattern)`
to
`new RegExp(this.options.patterns[pattern])`

will fix this issue.

**Comment by mheydman** [comment]
Regardless, it seems no matter what RegEx I use, I always get a message that says "Please match the requested format" whenever I attempt to submit my form.

**Comment by gakimball** [comment]
I have a pull request open that addresses this: #8313

Going to do a bit more testing to verify no existing functionality broke. Feel free to take a look. :)

**Comment by gakimball** [comment]
Merged #8313
